nautilus_trader/data/aggregation.pyx

```diff
@@ -1073,11 +1136,28 @@ cdef class TimeBarAggregator(BarAggregator):
             self._batch_post_update(ts_init)
 
     cpdef void _build_bar(self, TimeEvent event):
+        cdef int step = self.bar_type.spec.step
+
+        if self.bar_type.spec.aggregation != BarAggregation.MONTH:
+            # On receiving this event, timer should now have a new `next_time_ns`
+            self.next_close_ns = self._clock.next_time_ns(self._timer_name)
+        else:
+            alert_time = unix_nanos_to_dt(event.ts_event) + pd.DateOffset(months=step)
+
+            self._clock.set_time_alert(
+                name=self._timer_name,
+                alert_time=alert_time,
+                callback=self._build_bar,
+                override=True,
+            )
+
+            self.next_close_ns = dt_to_unix_nanos(alert_time)
+
         if not self._builder.initialized:
             # Set flag to build on next close with the stored close time
             # _build_on_next_tick is used to avoid a race condition between a data update and a TimeEvent from the timer
             self._build_on_next_tick = True
-            self._stored_close_ns = self.next_close_ns
+            self._stored_close_ns = event.ts_event
```

Comment:
I think this part of the logic is harder to reason about. It could be fixing a bug, especially if `self.next_close_ns` wasn't being updated -- but if it was, we're changing this from next close to current event time?

This was also the block we were considering removing? The comprehensive test coverage may help us to make a decision here.